modules/widgets/internet-defense-league.php

```diff
@@ -0,0 +1,163 @@
+<?php /*
+
+**************************************************************************
+
+Plugin Name:  Internet Defense League
+Description:  Displays your support for the Internet Defense League.
+Author:       Automattic Inc.
+Author URI:   https://automattic.com/
+
+**************************************************************************/
+
+class Jetpack_Internet_Defense_League_Widget extends WP_Widget {
+
+	public $defaults = array();
+
+	public $variant;
+	public $variants = array();
+
+	public $campaign;
+	public $campaigns  = array();
+	public $no_current = true;
+
+	public $badge;
+	public $badges = array();
+
+	function __construct() {
+		parent::__construct(
+			'internet_defense_league_widget',
+			/** This filter is documented in modules/widgets/facebook-likebox.php */
+			apply_filters( 'jetpack_widget_name_widget', esc_html__( 'Internet Defense League', 'jetpack' ) ),
+			array(
+				'description' => __( 'Show your support for the Internet Defense League.', 'jetpack' ),
+				'customize_selective_refresh' => true,
+			)
+		);
+
+		// When enabling campaigns other than 'none' or empty, change $no_current to false above.
+		$this->campaigns = array(
+			''       => __( 'All current and future campaigns', 'jetpack' ),
+			'none'   => __( 'None, just display the badge please', 'jetpack' ),
+		);
+
+		$this->variants = array(
+			'banner' => __( 'Banner at the top of my site', 'jetpack' ),
+			'modal'  => __( 'Modal (Overlay Box)', 'jetpack' ),
+		);
+
+		$this->badges = array(
+			'shield_badge'   => __( 'Shield Badge', 'jetpack' ),
+			'super_badge'    => __( 'Super Badge', 'jetpack' ),
+			'side_bar_badge' => __( 'Red Cat Badge', 'jetpack' ),
+		);
+
+		if ( $this->no_current === false ) {
+			$this->badges[ 'none' ] = __( 'Don\'t display a badge (just the campaign)', 'jetpack' );
```

Comment:
[Coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#space-usage) say whitespace only for variables in the bracket